### DISCUSSION TITLE
[@types/passport] User interface should have an optional id param instead of being empty

### DISCUSSION THREAD

**Discussion by franklygeorgy** [DefinitelyTyped/DefinitelyTyped#70922] [Issues with a @types package]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/93a782948d4ca1b68e686e01fe6d9e819aec2423/types/passport/index.d.ts#L8

Should probably be
```

interface User {
    id?: number
}

```
or
```

interface User {
    id?: any
}

```

This would make TSC happy when I follow the example at https://github.com/mjhea0/koa-passport-example which is linked from https://github.com/rkusa/koa-passport-example?tab=readme-ov-file#koa-router (`Example with koa-router`) which is linked from https://www.npmjs.com/package/koa-passport#usage (`Example Application`)

[a comment by an automated account is not shown]

**Comment by franklygeorgy** [comment]
Alternatively I would be fine if someone could help me figure out a way to override the def for the `User` interface to include `id` in my project, while still using the rest of the defs provided by this package

**Comment by graham-fisher-ii** [comment]
@franklygeorgy Came across this old issue - in case you're still wondering, the `User` type is defined as an `interface` to allow extension via declaration merging via module augmentation: https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation. In this case, the `Express` types are declared as `global` so we use [global augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#global-augmentation). Try adding this to your code:
```ts
global {
  namespace Express {
    interface User {
      id: number;
    }
  }
}
```
